### ISSUE TITLE
Assessment UI starts shifting to the left if you page up / down while entering an assessment - MIGRATED ISSUE

### ISSUE THREAD

**Issue by pbarry-r7** [rapid7/attackerkb#30]
Noticed while trying to paste in a lot of text into the assessment window, then hit page up / down in the assessment text box:

![image](https://user-images.githubusercontent.com/4108654/76256573-2ce57e80-621e-11ea-98a8-21a4de144bb2.png)

To reproduce, create an assessment, and past in the following text on https://attackerkb.com/topics/cve-2020-10189 using Google Chrome 80.

Then in the text box for the assessment, hit the page up / down button a few times. Notice that the page starts shifting left on its own until it's unreadable, and you have to reload the page to reset it.

Note: as @tekwizz123 points out, this does not appear to be a problem with Firefox.

```
MS-ISAC ADVISORY NUMBER:

2020-033 - UPDATED

 

DATE(S) ISSUED:

03/07/2020

03/09/2020 - UPDATED

 

SUBJECT:

A Vulnerability in ManageEngine Desktop Central Could Allow for Remote Code Execution

 

OVERVIEW:

A vulnerability in ManageEngine Desktop Central could allow for remote code execution. Desktop Central is a unified endpoint management solution that helps in managing servers, laptops, desktops, smartphones, and tablets from a central location. Successful exploitation of this vulnerability could allow for remote code execution within the context of SYSTEM. An attacker could then install programs; view, change, or delete data; or create new accounts with full user rights.

 

THREAT INTELLIGENCE:

Proof of Concept has been publicly released.

 

March 9 – UPDATED THREAT INTELLIGENCE:

This vulnerability has been observed and reported being actively exploited in the wild.

 

SYSTEM AFFECTED:

·  Desktop Central 10 prior to 10.0.474

 

RISK:

Government:

Large and medium government entities: High
Small government entities: Medium
Businesses:

Large and medium business entities: High
Small business entities: Medium
Home users: Low

 

TECHNICAL SUMMARY:

A vulnerability in ManageEngine Desktop Central could allow for remote code execution on affected installations (CVE-2020-10189). This issue occurs during deserialization of a specially crafted file due to improper input validation in the FileStorage class. An unauthenticated attacker can upload a malicious file and make a subsequent request to the uploaded file in order to trigger the malicious code.

 

Successful exploitation of this vulnerability could allow for remote code execution within the context of SYSTEM. An attacker could then install programs; view, change, or delete data; or create new accounts with full user rights.

 

RECOMMENDATIONS:

We recommend the following actions be taken:

Apply appropriate updates provided by ManageEngine to affected systems, immediately after appropriate testing.
Apply the Principle of Least Privilege to all systems and services.
Verify no unauthorized system modifications have occurred on system before applying patch.
Monitor intrusion detection systems for any signs of anomalous activity.
Unless required, limit external network access to affected products.
 

REFERENCES:

Source Incite:

https://srcincite.io/advisories/src-2020-0011/

 

ManageEngine:

https://www.manageengine.com/products/desktop-central/remote-code-execution-vulnerability.html

 

CVE:

https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-10189

 

24×7 Security Operations Center

Multi-State Information Sharing and Analysis Center (MS-ISAC)

Elections Infrastructure Information Sharing and Analysis Center (EI-ISAC)

31 Tech Valley Drive

East Greenbush, NY 12061

SOC@cisecurity.org - 1-866-787-4722

 

cid:image001.png@01D47C0B.665CCFB0cid:image002.jpg@01D47C0B.665CCFB0

cid:image003.png@01D47C0B.665CCFB0cid:image004.png@01D47C0B.665CCFB0cid:image005.png@01D47C0B.665CCFB0cid:image006.png@01D47C0B.665CCFB0

 

TLP: WHITE

Disclosure is not limited. Subject to standard copyright rules, TLP: WHITE information may be distributed without restriction.

http://www.us-cert.gov/tlp/

This message and attachments may contain confidential information. If it appears that this message was sent to you by mistake, any retention, dissemination, distribution or copying of this message and attachments is strictly prohibited. Please notify the sender immediately and permanently delete the message and any attachments.
```

**Comment by pbarry-r7** [comment]
"Which browser are you encountering this error in? I tried to replicate your issue in Firefox but was unsuccessful. Both attempting to write the issue using the text you provided, followed by pressing the Page Up and Page Down keys, as well as previewing the text, followed by pressing the Page Up and Page Down keys, failed to produce any change in how the text was laid out on the screen."

"I am using Chrome Version 80.0.3987.122 (Official Build) (64-bit)
The issue seems to appear when you hit the upper or lower limits of the text box and the whole page begins to scroll.

I just tested in Firefox, it seems that browser also doesn't have the issue where the whole page scrolls when you hit the top or bottom of the text box. Verified everything works as you'd expect there. Thanks for checking it out!"

"No problem, I'm curious if any other browsers have the same issue. I'll try Edge quickly so long, see if it causes any issues.

Confirmed this is not an issue on Microsoft Edge build 18363. There is what initially appears to be an issue as holding down the page down button will cause Edge to page down until the end of the input box and then loop back around to the top of the input box, however this is a relatively minor issue; no text misalignment occurs.

Note this is the legacy version of Edge, I have not tested with the Chromium based version of Edge (mostly because apparently you can't have both the legacy and the newer versions of Edge on the same PC....)

Ok upgraded to Microsoft Edge with Chromium and can confirm that on the latest build (downloaded from MS website today) this issue can be replicated, as shown below:

![Replication of BusterBs Issue](https://user-images.githubusercontent.com/2746189/76332068-eba4ab80-62bd-11ea-9709-9fe8e406161c.PNG)

Therefore it seems likely this is a Chromium engine related issue."

**Comment by pbarry-r7** [comment]
"fixed"

"Just tested this issue again in Microsoft Edge with Chromium and the root issue still seems to be there. Attached a screenshot below showing that the issue is still occurring as the text goes to the left of the page after hitting the Page Up and Page Down keys a few times.

Edit: Replaced the last image with one better representing the issue. Also the previous statement about the issue seeming to have been lessened turned out to not be true after further testing.

![issue still occuring in Microsoft Edge](https://user-images.githubusercontent.com/2746189/76691622-9ef10580-661a-11ea-8520-9dc11dc4d80f.PNG)
"

"Thanks for the verification! We've got a potential fix for this that hasn't been deployed yet. Looks like our git workflow marked this as closed prematurely. We'll update here when we push the fix live!"

"Alright, now the fix has been deployed. Let us know if you still see any issues related to this bug!"

"Thanks, can confirm that this appears to be fixed in my tests! :)"
